godot-macros/src/class/derive_godot_class.rs

```diff
@@ -742,6 +750,35 @@ fn parse_fields(
                 ));
             }
         } else {
+            if field.is_phantomvar {
+                if let Some(field_var) = &mut field.var {
+                    if field_var.getter.is_generated() {
+                        errors.push(error!(
+                            field_var.span,
+                            "PhantomVar<T> stores no data, so it cannot use an autogenerated getter"
+                        ));
+                    }
+                    if field_var.setter.is_generated() {
+                        errors.push(error!(
+                            field_var.span,
+                            "PhantomVar<T> stores no data, so it cannot use an autogenerated setter"
+                        ));
+                    }
+
+                    if field_var.getter.is_omitted() && field_var.setter.is_omitted() {
+                        errors.push(error!(
+                            field_var.span,
+                            "PhantomVar<T> is useless without a custom getter, setter, or both"
+                        ));
+                    }
+                } else {
+                    errors.push(error!(
+                        field.span,
+                        "PhantomVar<T> field is useless without attribute #[var]"
+                    ));
+                }
```

Comment:
Maybe extract into a function, e.g. `validate_phantomvar(...)`

Reply:
OK but then I'll also extract `validate_base_field` for the code right above this.